### ISSUE TITLE
yarn --pnp breaks nuxt

### ISSUE THREAD

**Issue by aparajita** [nuxt/nuxt.js#5008]
### Version

[v2.4.3](https://github.com/nuxt.js/releases/tag/v2.4.3)

### Reproduction link

[https://github.com/nuxt/nuxt.js/pull/4875](https://github.com/nuxt/nuxt.js/pull/4875)

### Steps to reproduce

create-nuxt-app test
[select yarn as package manager]
cd test
rm -rf ./node_modules
yarn --pnp
yarn run dev

### What is expected ?

I expect nuxt to run the dev server.

### What is actually happening?

/bin/sh: nuxt: command not found
error Command failed with exit code 127.

### Additional comments?

See https://github.com/nuxt/nuxt.js/pull/4875#issuecomment-462224664 for reference.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8642">#c8642</a>)</em></sub></div>

**Comment by pi0** [comment]
The first issue found. PNP seems not supporting `bin` field as an object. (Only string supported) workaround for testing: `yarn node -p "require('nuxt-edge/bin/nuxt')"`  (more issues exist) will update issue if found workarounds

**Comment by alexanderniebuhr** [comment]
Are there any updates on this? I really want to dive deep into Yarn Plug’n’Play.
Could you provide any workarounds for testing?

**Comment by aparajita** [comment]
Unfortunately until more people use modern package managers (that share a package store instead of copying every package over and over again into a flat hierarchy), this seems to be a non-issue for the maintainers. Please correct me @pi0 if I'm wrong.

If I understand correctly, the current stance is: module X depends on Y, and Y depends on Z, therefore X should be able to implicitly depend on Z. Unfortunately that is depending on an implementation detail of the current generation of package managers. And no package manager ever promised that dependencies are transitive in this way.

Honestly @pi0 I'm struggling to understand the reluctance to declare dependencies in the package in which they are required. Is it a philosophical thing? To say you want to modularize dependencies by pushing them up the module hierarchy doesn't quite make sense. No matter how many times you declare a dependency, if it's the same version, there will only be one copy installed, whether it's a flattened hierarchy or a shared store. So modularization of code !== modularization of dependencies.

**Comment by danielroe** [comment]
The [`bin` bug](https://github.com/nuxt/nuxt.js/issues/5008#issuecomment-462510739) identified by @pi0 appears to be fixed in yarn v2. (There are still undeclared dependencies within Nuxt.)

**Comment by pi0** [comment]
> So modularization of code !== modularization of dependencies.

Yeah, I completely agree with you @aparajita. The concern was that we splitted `@nuxt/vue-app` and `@nuxt/webpack` to make nuxt framework/bundler agnostic in the future. Like also supporting parcel and rollup and also react, etc. Those users don't need these default packages to be installed. 

BTW by the time, it seems that's not going to happen. Nuxt community and ecosystem are tightly coupled with both webpack and vue. I'll ensure this with core team and make a PR to add explicit dependencies.

**Comment by aparajita** [comment]
!@pi0 متشکرم
